Rubberduck.CodeAnalysis/CodePathAnalysis/Walker.cs

```diff
@@ -5,14 +5,23 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Linq;
+using Antlr4.Runtime;
 
 namespace Rubberduck.Inspections.CodePathAnalysis
 {
     public class Walker
     {
         public INode GenerateTree(IParseTree tree, Declaration declaration)
+        {
+            AssignmentNode lastAssignment = null;
+            bool isBranchCondition = false;
+            return GenerateTree(tree, declaration, ref lastAssignment, ref isBranchCondition);
+        }
+
+        public INode GenerateTree(IParseTree tree, Declaration declaration, ref AssignmentNode lastAssignment, ref bool isBranchCondition)
         {
             INode node = default;
+            VBAParser.BooleanExpressionContext branchCondition;
```

Comment:
Assigned, but not used - feels confusing & wrong

Reply:
Agreed. A proper solution would be to have a true expression evaluation engine (which is a open issue ATM as we have 3 engines, each not working 100% to specs) but that's definitely beyond this PR's scope.